### ISSUE TITLE
push image does not work with a filter

### ISSUE THREAD

**Issue by sabf81** [fabric8io/docker-maven-plugin#1350]
### Description
I try to push an image within my POM.
The Jenkins-Job log says the following:
```
08:45:23  [INFO] --- docker-maven-plugin:0.33.0:push (push-image) @ docker-flyway ---
08:45:23  [WARNING] DOCKER> None of the resolved images [edict/docker-flyway:4.2.0-9-SNAPSHOT-EOC-31375-poc-fabric8io-11] match the configured filter 'edict/docker-flyway:4.2.0-9-SNAPSHOT-EOC-31375-poc-fabric8io-11}'
```
Thats my POM content:

```
<build>
    <pluginManagement>
        <plugins>
            <plugin>
                <groupId>io.fabric8</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <version>0.33.0</version>
                <configuration>
                    <registry>${docker.registry.url}</registry>
                    <verbose>true</verbose>
                    <useColor>true</useColor>
                </configuration>
            </plugin>
        </plugins>
    </pluginManagement>
<plugins>
<plugin>
    <groupId>io.fabric8</groupId>
    <artifactId>docker-maven-plugin</artifactId>
    <executions>
        <execution>
            <id>build-and-tag-image</id>
            <phase>package</phase>
            <goals>
                <goal>build</goal>
            </goals>
            <configuration>
                <images>
                    <image>
                        <name>${docker-image.name}</name>
                        <build>
                            <contextDir>${project.build.directory}/dockerbuild</contextDir>

                            <tags>
                                <tag>${docker-image.version}</tag>
                                <tag>latest</tag>
                            </tags>
                        </build>
                    </image>
                </images>
            </configuration>
        </execution>
    </executions>
</plugin>
        </plugins>
        </build>
<profiles>
<profile>
    <id>edict-hudson-profile</id>
    <build>
        <plugins>
            <plugin>
                <groupId>io.fabric8</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>push-image</id>
                        <phase>deploy</phase>
                        <goals>
                            <goal>push</goal>
                        </goals>
                        <configuration>
                            <filter>${docker-image.name}:${docker-image.version}</filter>
                            <images>
                                <image>
                                    <name>${docker-image.name}:${docker-image.version}</name>
                                </image>
                            </images>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</profile>
</profiles>
```

**How can I get it work?**

### Info

* d-m-p version : 0.33.0
* Maven version (`mvn -v`) : 3.6.3
```

```
* Docker version :
```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

```

**Comment by sabf81** [comment]
I did not understand this sentence correctly:
> This goal uploads images to the registry which have a <build> configuration section.

I have to add a <build> Tag in my configuration part too, to get it pushed.
Now my POM looks like that:

```
<plugin>
                <groupId>io.fabric8</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>build-and-tag-image</id>
                        <phase>package</phase>
                        <configuration>
                            <images>
                                <image>
                                    <name>${docker-image.name}</name>
                                    <build>
                                        <contextDir>${project.build.directory}/dockerbuild</contextDir>

                                        <tags>
                                            <tag>${docker-image.version}</tag>
                                            <tag>latest</tag>
                                        </tags>
                                    </build>
                                </image>
                            </images>
                        </configuration>
                        <goals>
                            <goal>build</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
    <profiles>
        <profile>
            <id>edict-hudson-profile</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>io.fabric8</groupId>
                        <artifactId>docker-maven-plugin</artifactId>

                        <executions>
                            <execution>
                                <id>push-and-remove-local-image</id>
                                <phase>deploy</phase>
                                <configuration>
                                    <removeMode>all</removeMode>
                                    <images>
                                        <image>
                                            <name>${docker-image.name}</name>
                                            <build>
                                                <tags>
                                                    <tag>${docker-image.version}</tag>
                                                </tags>
                                            </build>
                                        </image>
                                    </images>
                                </configuration>
                                <goals>
                                    <goal>push</goal>
                                    <goal>remove</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
                </plugins>
            </build>
        </profile>
    </profiles>
```

**Comment by rhuss** [comment]
> ```
> 08:45:23  [WARNING] DOCKER> None of the resolved images [edict/docker-flyway:4.2.0-9-SNAPSHOT-EOC-31375-poc-fabric8io-11] match the configured filter 'edict/docker-flyway:4.2.0-9-SNAPSHOT-EOC-31375-poc-fabric8io-11}'
> ```

It looks like your filter had an extra `}` at the end. Not sure where it comes from, maybe you used a different pom than poster ?
